### ISSUE TITLE
eb deploy exiting with exit code 0 even when ServiceError

### ISSUE THREAD

**Issue by dortort** [gxela/awsebcli#12]
As mentioned here: https://forums.aws.amazon.com/thread.jspa?messageID=715094

"ServiceError", due to an environment in "invalid state", returns nonetheless exit code 0.

This is an issue especially for CI/ CD setups.

**Comment by vergenzt** [comment]
I would bet that this is caused by [this line](https://github.com/gxela/awsebcli/blob/master/ebcli/core/ebcore.py#L141):

```python
def main():
    # ...

    app = EB()

    try:
        app.setup()
        app.run()

    # ... various except clauses with various useful return codes

    finally:
        app.close()
```

I think that last `app.close()` causes the app to always exit 0, since `finally` clauses are guaranteed to run.

I've proposed a fix at #15.

**Comment by vergenzt** [comment]
See @rohitkothari's [comment on #15](https://github.com/gxela/awsebcli/pull/15#issuecomment-221632008). Seems to provide a good explanation of current situation if anyone's seeing this later.

**TL;DR**: This repo seems to only have up to version 3.0.10, whereas the default `pip install` version is 3.7.6. This bug is present in the latter, but not in the former.

**Comment by vergenzt** [comment]
This issue is causing my CI builds to mark as passed when the deployment fails.

**Comment by vergenzt** [comment]
FWIW the bug appears to have been introduced by version 3.8.4.

(3.8.3 returns an error code upon failure; 3.8.4 does not.)
